src/renderer/components/hotbar/hotbar-menu.tsx

```diff
@@ -38,36 +35,67 @@ export class HotbarMenu extends React.Component<Props> {
     return item ? catalogEntityRegistry.items.find((entity) => entity.metadata.uid === item.entity.uid) : null;
   }
 
-  previous() {
-    HotbarStore.getInstance().switchToPrevious();
-  }
+  onDragEnd(result: DropResult) {
+    const { source, destination } = result;
 
-  next() {
-    HotbarStore.getInstance().switchToNext();
-  }
+    if (!destination) {  // Dropped outside of the list
+      return;
+    }
 
-  openSelector() {
-    CommandOverlay.open(<HotbarSwitchCommand />);
+    const from = parseInt(source.droppableId);
+    const to = parseInt(destination.droppableId);
+
+    HotbarStore.getInstance().restackItems(from, to);
   }
 
   renderGrid() {
-    if (!this.hotbar.items.length) return;
-
     return this.hotbar.items.map((item, index) => {
       const entity = this.getEntity(item);
 
       return (
-        <HotbarCell key={index} index={index}>
-          {entity && (
-            <HotbarIcon
-              key={index}
+        <Droppable droppableId={`${index}`} key={index}>
+          {(provided, snapshot) => (
+            <HotbarCell
               index={index}
-              entity={entity}
-              isActive={this.isActive(entity)}
-              onClick={() => entity.onRun(catalogEntityRunContext)}
-            />
+              key={entity ? entity.getId() : `cell${index}`}
+              innerRef={provided.innerRef}
+              className={cssNames({ isDraggingOver: snapshot.isDraggingOver })}
+              {...provided.droppableProps}
+            >
+              {entity && (
+                <Draggable draggableId={item.entity.uid} key={item.entity.uid} index={0}>
+                  {(provided, snapshot) => {
+                    const style = {
+                      zIndex: defaultHotbarCells - index,
```

Comment:
Needed for moving cell effect - old cell must cover an empty one.
![cell zindex](https://user-images.githubusercontent.com/9607060/116844975-464d4500-abed-11eb-8fe3-33e07f6848a9.gif)